gravitee-apim-console-webui/src/management/settings/groups/group/edit-member-dialog/edit-member-dialog.component.ts

```diff
@@ -0,0 +1,291 @@
+          next: (response) => {
+            const reference = response.find((u) => u.id === currentPrimaryOwner.id).reference;
+            const ownerMembership = this.mapGroupMembership(currentPrimaryOwner.id, reference, RoleName.OWNER);
+            const primaryOwnership = this.mapGroupMembership(this.user.id, this.user.reference, RoleName.PRIMARY_OWNER);
+            this.updatedMemberships = [ownerMembership, primaryOwnership];
+            this.ownershipTransferMessage = `${currentPrimaryOwner.displayName} is the API primary owner. The primary ownership will be transferred to ${this.member.displayName} and ${currentPrimaryOwner.displayName} will be updated as owner.`;
+          },
+        });
+      }
+    } else if (this.isDowngradeRole()) {
+      if (this.selectedPrimaryOwner) {
+        this.usersService.search(this.selectedPrimaryOwner.displayName).subscribe({
+          next: (response) => {
+            const reference = response.find((user) => user.id === this.selectedPrimaryOwner.id).reference;
+            const groupMembership = this.mapGroupMembership(
+              this.user.id,
+              this.user.reference,
+              this.editMemberForm.controls['defaultAPIRole'].value,
+            );
+            const primaryOwnership = this.mapGroupMembership(this.selectedPrimaryOwner.id, reference, RoleName.PRIMARY_OWNER);
+            this.updatedMemberships = [groupMembership, primaryOwnership];
+            this.ownershipTransferMessage = `${this.member.displayName} is the API primary owner. The primary ownership will be transferred to ${this.selectedPrimaryOwner.displayName} and ${this.member.displayName} will be updated as owner.`;
+          },
+          error: () => {
+            this.snackBarService.error(`Error occurred while fetching user details.`);
+          },
+        });
+      }
+    } else {
+      let groupMembership = this.updatedMemberships.find((membership) => membership.id === this.user.id);
+
+      if (!groupMembership) {
+        groupMembership = this.mapGroupMembership(this.user.id, this.user.reference, this.editMemberForm.controls['defaultAPIRole'].value);
+      }
+      this.updatedMemberships = [groupMembership];
+    }
+  }
+
+  mapGroupMembership(id: string, reference: string, defaultAPIRole: string): GroupMembership {
+    return {
+      id: id,
+      reference: reference,
+      roles: [
+        {
+          name: defaultAPIRole,
+          scope: 'API',
+        },
+        {
+          name: this.editMemberForm.controls['defaultApplicationRole'].value,
+          scope: 'APPLICATION',
+        },
+        {
+          name: this.editMemberForm.controls['defaultIntegrationRole'].value,
+          scope: 'INTEGRATION',
+        },
+      ],
+    };
+  }
+
+  selectPrimaryOwner($event: MatAutocompleteSelectedEvent) {
+    this.selectedPrimaryOwner = $event.option.value;
+    this.editMemberForm.controls['searchTerm'].setValue('');
```

Comment:
I think that if the form is typed you should be able to access to the controls directly without using the array.

```suggestion
    this.editMemberForm.controls.searchTerm.reset();
```